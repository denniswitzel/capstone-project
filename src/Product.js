import React from 'react'

export default function Product({ title, image }) {
    return (
    <>
    <h2>{title}</h2>
    <img src={image} alt={title}/>
    </>
    )
}