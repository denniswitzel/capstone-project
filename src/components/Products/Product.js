import React from 'react'
import styled from 'styled-components'
import { ReactComponent as Star } from '../../images/icons/star.svg'
import PropTypes from 'prop-types'

Product.propTypes = {
  title: PropTypes.string,
  image: PropTypes.string,
  rating: PropTypes.string,
  onClick: PropTypes.func
}

export default function Product({ title, image, rating, onClick }) {
  return (
    <ProductWrapper onClick={onClick}>
      <ProductName>{title}</ProductName>
      <ProductImage src={image} alt={title} />
      <RatingWrapper>
      <StyledRating>{rating}</StyledRating>
      {rating && <StyledStar />}
      </RatingWrapper>
    </ProductWrapper>
  )
}

const ProductWrapper = styled.div`
  background: linear-gradient(45deg, var(--turquoise), var(--green));
  border-radius: 20px 0 20px 20px;
  width: 151px;
  height: 143px;
  box-shadow: 2px 2px 6px rgba(0, 0, 0, 0.2);
  padding-top: 15px;
  display: inline-block;
  margin: 25px 15px 50px 15px;
`

const ProductName = styled.h3`
  color: var(--white);
  font-family: 'Open Sans', sans-serif;
  font-weight: 300;
  font-size: 14px;
  width: 80%;
  height: 60px;
  text-align: center;
  margin: 0 auto;
`
const ProductImage = styled.img`
  width: 75%;
  margin-left: 15px;
  margin-top: 10px;
  -webkit-filter: drop-shadow(2px 2px 6px rgba(0, 0, 0, 0.2));
  filter: drop-shadow(2px 2px 6px rgba(0, 0, 0, 0.2));
`

const RatingWrapper = styled.div`
  display: flex;
  align-items: end;
  justify-content: center;
  margin-top: 5px;  
`
const StyledRating = styled.p`
    font-size: 14px;
    color: var(--grey);
    margin: 0 0 0 10px;  
    float: left;
`

const StyledStar = styled(Star)`
  fill: var(--green);
  height: 20px;
  float: left;
`