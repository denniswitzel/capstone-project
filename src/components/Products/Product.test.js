import React from 'react'
import renderer from 'react-test-renderer'
import Product from './Product'
import { render } from '@testing-library/react'

describe('Product', () => {
    const title = 'Like meat - Like Nuggets'
    const image = 'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/nuggets/Like-meat-Nuggets.png'

    it('display title and image', () => {
        const { getByText } = render(<Product title={title} image={image} key={title}/>)
        expect(getByText(title)).toBeInTheDocument()
        expect(document.querySelector("img").getAttribute("src")).toBe(image)
    })

    it('renders correctly', () => {
        const tree = renderer.create(<Product />)
        expect(tree).toMatchSnapshot()
      })
  })

