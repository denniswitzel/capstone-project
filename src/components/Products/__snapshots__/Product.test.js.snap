// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
<div
  className="sc-AxjAm chUUcj"
>
  <h3
    className="sc-AxirZ iyHzja"
  />
  <img
    className="sc-AxiKw kFMpIF"
  />
  <div
    className="sc-AxhCb gCPGaY"
  >
    <p
      className="sc-AxhUy cKGfdj"
    />
  </div>
</div>
`;
